provider/datagen/src/bin/icu4x-datagen.rs

```diff
@@ -132,34 +132,17 @@ struct Cli {
     )]
     include_collations: Vec<CollationTable>,
 
-    #[arg(long, hide = true)]
-    #[arg(help = "Deprecated, use --locales full or --locales modern")]
-    cldr_locale_subset: bool,
-
     #[arg(long, short, num_args = 1..)]
     #[arg(
         help = "Include these resource keys in the output. Accepts multiple arguments.\n\
                 Set to 'all' for all keys, or 'none' for no keys."
     )]
     keys: Vec<String>,
 
-    #[arg(long, value_name = "KEY_FILE")]
-    #[arg(
-        help = "Path to text file with resource keys to include, one per line. Empty lines \
-                  and lines starting with '#' are ignored.\n
-                  Requires the `legacy_api` Cargo feature."
-    )]
-    #[cfg(feature = "legacy_api")]
-    key_file: Option<PathBuf>,
-
     #[arg(long, value_name = "BINARY")]
     #[arg(help = "Analyzes the binary and only includes keys that are used by the binary.")]
     keys_for_bin: Option<PathBuf>,
 
-    #[arg(long, hide = true)]
-    #[arg(help = "Deprecated: alias for --keys all")]
-    all_keys: bool,
-
```

Comment:
Nit: We've been piloting the new names for a while, but I still use `--all-keys --all-locales` because I find them easier to remember than `--keys all --locales full`. Please pull these two specific removals to their own PR